### ISSUE TITLE
Java 23 Facet

### ISSUE THREAD

**Issue by jason-faust** [eclipse-wtp-common/webtools.common#11]
Could the Java 23 facet be be added to work with the 2024-12 simultaneous release?

**Comment by robstryker** [comment]
My understanding is that JDT doesn't really have a fully working java-23 support yet... but I may be slightly out of the loop or outdated in my assumption.

**Comment by jason-faust** [comment]
It does as far as I can tell. There wasn't really an announcement in the news and noteworthy, but the Java 23 target is available in the IDE for basic Java projects (as well as m2e), and ECJ 3.40 can emit Java 23 byte code.

As a test I mimicked the contents of a7a1ecece2298e547e88f996e78f46d3772d2eb4 and I was able to generate the plug-in without a build error. Just could not figure out how to get it to load into a copy of 2024-12 by replacing the jar in the plugins, features, or dropins folders.

**Comment by nitind** [comment]
> Could the Java 23 facet be be added to work with the 2024-12 simultaneous release?

No, once it's out the door the focus becomes the next release. It's in 2025-03 (tomorrow's release), along with version 24. As Rob alluded to, our having the facet version number available is not tied to JDT, but support for the version itself is all handled by them.
